client/src/components/Search/index.js

```diff
@@ -59,29 +58,27 @@ export default class Search extends Component {
    *
    * @param {*} placeDetail 
    */
+  
   onPlaceFound(placeDetail) {
     this.props.map.setZoom(16);
     const location = placeDetail.geometry.location;
-    const userMarker = L.marker(location).addTo(this.props.map)
+    
+    if (marker) 
```

Comment:
use `this.state.marker` in place of the `marker` variable throughout this function.

Reply:
Since we are managing the state in the parent,i.e, LeafletOSMMap, we then need to do it there? We can't change the state in the child right, only affect the changes through prop. I am new to React so, correct me if I am wrong or is there another correct pattern to do this?
Initially I intended to do all the marker stuff in parent component of Search so that I could affect the state, but based on the comment I figured we need to follow this approach.

Reply:
this article might be helpful https://medium.freecodecamp.org/where-do-i-belong-a-guide-to-saving-react-component-data-in-state-store-static-and-this-c49b335e2a00

Reply:
Redux is the most preferred way to handle store. Currently we don't use Redux in the project and redux makes me dizzy, the learning curve there with react feels too much for me to handle at present, that could be just me 😬 . I am comfortable with React's local store usage, in case we move to Redux I am back to reading.

Reply:
Redux would be new to me too but I don't think we need it in this case...will read up on it this weekend to confirm. But for this, each component can have its own state and consume props passed to it. See line 10 - we already have Search component specific state, add marker key to it for tracking Search result marker instance.

Reply:
ok @jkwening will make changes to hold it in the search component state.

Reply:
the dizzy 💫 feeling with redux is completely normal ... we don't need to go that deep just yet 🤔